agent/indy_catalyst_agent/conductor.py

```diff
@@ -111,7 +108,14 @@ def __init__(
 
         wallet_seed = self.settings.get("wallet.seed")
         public_did_info = await context.wallet.get_public_did()
-        if not public_did_info:
+
+        if public_did_info and seed_to_did(wallet_seed) != public_did_info.did:
+            raise StartupError(
+                f"New seed '{wallet_seed}' provided "
+                + f"but public seed is already set to {public_did_info.did}"
+            )
+
+        if wallet_seed:
```

Comment:
```suggestion
        if not public_did_info:
```
Although not sure what happens above if wallet_seed is blank.